ui/app/src/components/DatasourceList/DatasourceEditorForm.tsx

```diff
@@ -134,7 +134,7 @@ export function DatasourceEditorForm<T extends Datasource>(props: DatasourceEdit
               label="Name"
               value={state.metadata.name}
               InputProps={{
-                readOnly: isReadonly,
+                readOnly: saveActionStr !== 'Create',
```

Comment:
Not linked with this PR but the saveActionStr could be a limited string.
It would prevent from a developer to create the component with ``saveActionStr='new'`` for example, and would be more future proof

Reply:
e.g the `action` prop in `VariableFormDrawer`

Reply:
Oh didnt check I thought it was already done, let me check